### ISSUE TITLE
maximize the output of a built neural net( by pytorch) with continuous and discrete/integer variables

### ISSUE THREAD

**Issue by AmosJoseph** [ahmedfgad/GeneticAlgorithmPython#220]
Hi, can this library be used to maximize the output of a well-built neural net( by pytorch) with continuous and discrete/integer (1, 2, 3 ) variables?

Is there any example?

Best!

**Comment by ahmedfgad** [comment]
It can train models built in Keras and PyTorch.

Here you can find examples for training PyTorch models: https://github.com/ahmedfgad/GeneticAlgorithmPython/tree/master/examples/TorchGA

**Comment by AmosJoseph** [comment]
Thank you for your reply!
The neural net has been trained and tested with good accuracy.


Then how to maximize the output of the neural net in the input domain?



发自我的iPhone


------------------ Original ------------------
From: Ahmed Gad ***@***.***&gt;
Date: Fri,Jul 14,2023 10:43 PM
To: ahmedfgad/GeneticAlgorithmPython ***@***.***&gt;
Cc: AmosJoseph ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [ahmedfgad/GeneticAlgorithmPython] maximize the output of a builtneural net( by pytorch) with continuous and discrete/integer variables (Issue#220)

**Comment by ahmedfgad** [comment]
Do you you mean change the inputs (not the neural network itself) so that their outputs are accurate? If this is the case, then yes it can be done. Although there is no example yet, it can be developed easily.

**Comment by AmosJoseph** [comment]
Yes.&nbsp;Find the inputs (not the neural network itself) so that their outputs are maximum.


Then how to implement this?


Best!

发自我的iPhone


------------------ Original ------------------
From: Ahmed Gad ***@***.***&gt;
Date: Fri,Jul 14,2023 10:55 PM
To: ahmedfgad/GeneticAlgorithmPython ***@***.***&gt;
Cc: AmosJoseph ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [ahmedfgad/GeneticAlgorithmPython] maximize the output of a built neural net( by pytorch) with continuous and discrete/integer variables (Issue #220)





 
Do you you mean change the inputs (not the neural network itself) so that their outputs are accurate? If this is the case, then yes it can be done. Although there is no example yet, it can be developed easily.

**Comment by ahmedfgad** [comment]
Sorry for the long time to follow up. Just decided to reply in case it would still be helpful even for someone else.

Then what evolves is the inputs. Here are the steps:
1. Set the `num_genes` parameter to the length of a single input.
2. Inside the fitness function, pass the input to the neural network to return the predicted output.
3. Calculate the fitness by comparing the predicted and desired outputs.
